dialer.go

```diff
@@ -18,6 +18,13 @@ type Dialer struct {
 	// Unique identifier for client connections established by this Dialer.
 	ClientID string
 
+	// Optionally specifies the function that the dialer uses to establish
+	// network connections. If nil, net.(*Dialer).DialContext is used instead.
+	//
+	// When DialFunc is set, LocalAddr, DualStack, FallbackDelay, and KeepAlive
+	// are ignored.
+	DialFunc func(context.Context, string, string) (net.Conn, error)
```

Comment:
```suggestion
	DialFunc func(context context.Context, network string, address string) (net.Conn, error)
```
:point_up: suggest adding the names of arguments in case you don't have the `DialContext` on top of your head.